components/ArchiveList.php

```diff
@@ -0,0 +1,98 @@
+
+use DB;
+use Cms\Classes\Page;
+use Cms\Classes\ComponentBase;
+use RainLab\Blog\Models\Post as BlogPost;
+
+class ArchiveList extends ComponentBase
+{
+    /**
+     * A collection of the past x months
+     * @var Collection
+     */
+    public $archiveMonths;
+
+    /**
+     * @var string Reference to the page name for linking to archive.
+     */
+    public $archivePage;
+
+    /**
+     * @var string Reference to the current archive slug.
+     */
+    public $currentArchiveSlug;
+
+    public function componentDetails()
+    {
+        return [
+            'name'        => 'rainlab.blog::lang.settings.archive_list_name',
+            'description' => 'rainlab.blog::lang.settings.archive_list_description'
+        ];
+    }
+
+    public function defineProperties()
+    {
+        return [
+            'monthsToShow' => [
+                'title'             => 'rainlab.blog::lang.settings.archive_list_months_to_show',
+                'type'              => 'string',
+                'validationPattern' => '^[0-9]+$',
+                'validationMessage' => 'rainlab.blog::lang.settings.archive_list_months_to_show_validation',
+                'default'           => '5',
+            ],
+            'showPostCount' => [
+                'title'             => 'rainlab.blog::lang.settings.archive_list_show_post_count',
+                'type'              => 'checkbox',
+                'default'           => 'false'
+            ],
+            'slug' => [
+                'title'             => 'rainlab.blog::lang.settings.archive_list_slug',
+                'description'       => 'rainlab.blog::lang.settings.archive_list_slug_description',
+                'default'           => '{{ :slug }}',
+                'type'              => 'string'
+            ],
+            'archivePage' => [
+                'title'             => 'rainlab.blog::lang.settings.archive_list_page',
+                'description'       => 'rainlab.blog::lang.settings.archive_list_page_description',
+                'type'              => 'dropdown',
+                'default'           => 'blog/archive',
+                'group'             => 'Links',
+            ],
+        ];
+    }
+
+    public function getArchivePageOptions()
+    {
+        return Page::sortBy('baseFileName')->lists('baseFileName', 'baseFileName');
+    }
+
+    public function onRun()
+    {
+        $this->archiveMonths = $this->page['archiveMonths'] = $this->loadMonths($this->property('monthsToShow'));
+        $this->currentArchiveSlug = $this->page['currentArchiveSlug'] = $this->property('slug');
+        $this->archivePage = $this->page['archivePage'] = $this->property('archivePage');
+    }
+
+    protected function loadMonths($monthsToShow)
+    {
+        if (($monthsToShow > 0) ? $limit = $monthsToShow : $limit = 100);
+
+        $archiveRange = BlogPost::select(DB::raw("DATE_FORMAT(published_at, '%M %Y') AS month, count(*) as count"))
```

Comment:
Could you explain more about why you're only loading the months and post counts in this component instead of loading the relevant posts within each month? Not against it, just would like more information on the decision :)